src/hotspot/share/classfile/verifier.cpp

```diff
@@ -2038,7 +2038,8 @@ void ClassVerifier::verify_cp_type(
 
   verify_cp_index(bci, cp, index, CHECK_VERIFY(this));
   unsigned int tag = cp->tag_at(index).value();
-  if ((types & (1 << tag)) == 0) {
+  // Resolution errors start with JVM_CONSTANT_InternalMin = 100, which is not valid for shift op
+  if (tag >= JVM_CONSTANT_InternalMin || (types & (1 << tag)) == 0) {
```

Comment:
```suggestion
  if (tag > JVM_CONSTANT_ExternalMax || (types & (1 << tag)) == 0) {
```
This excludes the possibility of having a corrupt/bad tag > 32 but < 100